### DISCUSSION TITLE
websocket 1009 message too big

### DISCUSSION THREAD

**Discussion by xczhanjun** [coder/code-server#5278] [Q&A]
I want to open a big (>15M) jupyter notebook ,the get the error msg: websocket err 1009 message too big

**Comment by jsjoeio** [comment]
🤔 I honestly don't know how you get around this. Maybe @code-asher knows?

**Reply by code-asher** [reply]
Hmm this probably needs fixing upstream, I guess they need to chunk large messages or something.

**Comment by MiXaiLL76** [comment]
![изображение](https://github.com/user-attachments/assets/c6fed3ad-fa44-4807-b19c-24b9206df685)
Is this problem still not solved? My jupyter-notebook weighed 12mb and does not boot.

**Reply by code-asher** [reply]
We never looked into it, I am not sure if it is a problem with code-server or VS Code, would be helpful if anyone can reproduce with Codespaces to see if it happens there as well.

**Reply by MiXaiLL76** [reply]
Hi yes, actually, I just ran code-server inside jupyter for my work.

The trick is that jupyter, namely tornado https://github.com/tornadoweb/tornado/pull/3453 limits the messages that are transmitted via websockets.

And in the case of code-server, large files (for example, a large ipynb) are transmitted not in "chunks", but "whole".
i.e. 1 message consists of 10 or more megabytes at once. This is the problem.

There are two solutions:
1) Make a "hole" on the tornado side in websockets - a bigger one
2) Make a split of large objects into "chunks" on your side

**Comment by benz0li** [comment]
@xczhanjun @MiXaiLL76 I simply overwrite the default values when building the image.

See https://github.com/b-data/jupyterlab-python-docker-stack/blob/df9ebe2c0f0802718e2ab96ea1c76d3cba092358/base/latest.Dockerfile#L276-L283

ℹ  Python is installed at `/usr/local`
